include/seqan3/io/alignment_file/output.hpp

```diff
@@ -0,0 +1,940 @@
+     * ### Exceptions
+     *
+     * Basic exception safety.
+     *
+     * ### Example
+     *
+     * ```cpp
+     * alignment_file_out fout{"/tmp/my.fasta"};
+     *
+     * std::vector<std::tuple<dna5_vector, std::string>> range
+     * {
+     *     { "ACGT"_dna5, "First" },
+     *     { "NATA"_dna5, "2nd" },
+     *     { "GATA"_dna5, "Third" }
+     * }; // a range of "records"
+     *
+     * fout = range; // will iterate over the records and write them
+     * ```
+     */
+    template <typename rng_t>
+    alignment_file_out & operator=(rng_t && range)
+        requires std::ranges::InputRange<rng_t> && tuple_like_concept<reference_t<rng_t>>
+    {
+        for (auto && record : range)
+            push_back(std::forward<decltype(record)>(record));
+    }
+
+    /*!\brief            Write a range of records (or tuples) to the file.
+     * \tparam rng_t     Type of the range, must satisfy seqan3::std::ranges::InputRange and have a reference type that
+     *                   satisfies seqan3::tuple_like_concept.
+     * \param[in] range  The range to write.
+     * \param[in] f      The file being written to.
+     *
+     * \details
+     *
+     * This operator enables alignment_file_out to be at the end of a piping operation. It just calls
+     * operator=() internally.
+     *
+     * ### Complexity
+     *
+     * Linear in the number of records.
+     *
+     * ### Exceptions
+     *
+     * Basic exception safety.
+     *
+     * ### Example
+     *
+     * ```cpp
+     * alignment_file_out fout{"/tmp/my.fasta"};
+     *
+     * std::vector<std::tuple<dna5_vector, std::string>> range
+     * {
+     *     { "ACGT"_dna5, "First" },
+     *     { "NATA"_dna5, "2nd" },
+     *     { "GATA"_dna5, "Third" }
+     * }; // a range of "records"
+     *
+     * range | fout;
+     * // the same as:
+     * fout = range;
+     * ```
+     *
+     * This is especially useful in combination with file-based filters:
+     *
+     * ```cpp
+     * alignment_file_in{"input.fastq"} | view::minimum_average_quality_filter(20)
+     *                                 | view::minimum_alignment_length_filter(50)
+     *                                 | ranges::view::take(5)
+     *                                 | alignment_file_out{"output.fasta"};
+     * ```
+     */
+    template <typename rng_t>
+    friend alignment_file_out & operator|(rng_t && range, alignment_file_out & f)
+        requires std::ranges::InputRange<rng_t> && tuple_like_concept<reference_t<rng_t>>
+    {
+        f = range;
+        return f;
+    }
+
+    //!\overload
+    template <typename rng_t>
+    friend alignment_file_out operator|(rng_t && range, alignment_file_out && f)
+        requires std::ranges::InputRange<rng_t> && tuple_like_concept<reference_t<rng_t>>
+    {
+        f = range;
+        return std::move(f);
+    }
+    //!\}
+
+    /*!\name Tuple interface
+     * \brief Provides functions for field-based ("column"-based) writing.
+     * \{
+     */
```

Comment:
I don't think we really need the column-wise interface for alignment-files. It makes sense mostly only for sequence_files...

Reply:
does this consider blast files? I agree that SAM records, bloated as they are, are probably never used in column-based reading/writing. I don't know about blast so I just wanted to check.

Reply:
I don't think it's relevant for BLAST. And if it turns out that it is, we can still add it later.